core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -136,16 +136,16 @@ class UnifiedLog(@volatile var logStartOffset: Long,
    */
   @volatile private var firstUnstableOffsetMetadata: Option[LogOffsetMetadata] = None
 
+  @volatile var partitionMetadataFile: Option[PartitionMetadataFile] = None
+
+  @volatile private[kafka] var _localLogStartOffset: Long = logStartOffset
+
   /* Keep track of the current high watermark in order to ensure that segments containing offsets at or above it are
    * not eligible for deletion. This means that the active segment is only eligible for deletion if the high watermark
    * equals the log end offset (which may never happen for a partition under consistent load). This is needed to
    * prevent the log start offset (which is exposed in fetch responses) from getting ahead of the high watermark.
    */
-  @volatile private var highWatermarkMetadata: LogOffsetMetadata = new LogOffsetMetadata(logStartOffset)
-
-  @volatile var partitionMetadataFile: Option[PartitionMetadataFile] = None
-
-  @volatile private[kafka] var _localLogStartOffset: Long = logStartOffset
+  @volatile private var highWatermarkMetadata: LogOffsetMetadata = new LogOffsetMetadata(_localLogStartOffset)
```

Comment:
There won't be any effect with this change as `_localLogStartOffset` is initialized with `logStartOffset`. But it is good to keep `_localLogStartOffset` for consistency and relevance of this field.